src/Form/Type/AdminType.php

```diff
@@ -107,10 +108,7 @@ static function (array $associationMapping): string {
                         $subject = $p->getValue($parentSubject, $parentPath.$path);
                     } catch (NoSuchIndexException $e) {
                         // no object here, we create a new one
-                        // NEXT_MAJOR: Remove the null check.
-                        if (null !== $this->adminHelper) {
-                            $subject = $this->adminHelper->addNewInstance($parentSubject, $parentFieldDescription);
-                        }
+                        $subject = ObjectManipulator::setObject($admin->getNewInstance(), $parentSubject, $parentFieldDescription);
```

Comment:
@VincentLanglet There is BC break here, parent object is set to new instance instead of adding new instance to parent collection, and this does not work for me

Reply:
That's intended, to solve https://github.com/sonata-project/SonataAdminBundle/issues/6166
And it works well for me since month.

If you think a special case is missed create an issue with a precise description.
But without any information, I would say that your setter or mapping has something wrong.